applications/DEMApplication/python_scripts/DEM_analysis_stage.py

```diff
@@ -0,0 +1,647 @@
+import time as timer
+import os
+import sys
+from KratosMultiphysics import *
+from KratosMultiphysics.DEMApplication import *
+sys.path.insert(0, '')
+from analysis_stage import AnalysisStage
+
+# Import MPI modules if needed. This way to do this is only valid when using OpenMPI. For other implementations of MPI it will not work.
+if "OMPI_COMM_WORLD_SIZE" in os.environ or "I_MPI_INFO_NUMA_NODE_NUM" in os.environ:
+    if "DO_NOT_PARTITION_DOMAIN" in os.environ:
+        Logger.PrintInfo("DEM", "Running under MPI........")
+        from KratosMultiphysics.mpi import *
+        import DEM_procedures_mpi_no_partitions as DEM_procedures
+        import DEM_material_test_script
+    else:
+        Logger.PrintInfo("DEM", "Running under OpenMP........")
+        from KratosMultiphysics.MetisApplication import *
+        from KratosMultiphysics.MPISearchApplication import *
+        from KratosMultiphysics.mpi import *
+        import DEM_procedures_mpi as DEM_procedures
+        import DEM_material_test_script_mpi as DEM_material_test_script
+else:
+    Logger.PrintInfo("DEM", "Running under OpenMP........")
+    import DEM_procedures
+    import DEM_material_test_script
+
+class DEMAnalysisStage(AnalysisStage):
+
+    def GetParametersFileName(self):
+        return "ProjectParametersDEM.json"
+
+    def GetInputParameters(self):
+        parameters_file_name = self.GetParametersFileName()
+        parameters_file = open(parameters_file_name, 'r')
+        return Parameters(parameters_file.read())
+
+    def LoadParametersFile(self):
+        self.DEM_parameters = self.GetInputParameters()
+        self.project_parameters = self.GetInputParameters()
+        default_input_parameters = self.GetDefaultInputParameters()
+        self.DEM_parameters.ValidateAndAssignDefaults(default_input_parameters)
+
+    @classmethod
+    def GetDefaultInputParameters(self):
+        import dem_default_input_parameters
+        return dem_default_input_parameters.GetDefaultInputParameters()
+
+    @classmethod
+    def model_part_reader(self, modelpart, nodeid=0, elemid=0, condid=0):
+        return ReorderConsecutiveFromGivenIdsModelPartIO(modelpart, nodeid, elemid, condid, IO.SKIP_TIMER)
+
+    @classmethod
+    def GetMainPath(self):
+        return os.getcwd()
+
+    def __init__(self, model, parameters):
+        self.model = model
+        self.main_path = self.GetMainPath()
+        self.LoadParametersFile()
+        self.solver_strategy = self.SetSolverStrategy()
+        self.creator_destructor = self.SetParticleCreatorDestructor()
+        self.dem_fem_search = self.SetDemFemSearch()
+        self.procedures = self.SetProcedures()
+        self.SetAnalyticParticleWatcher()
+        self.PreUtilities = PreUtilities()
+        self.aux = AuxiliaryUtilities()
+
+        # Set the print function TO_DO: do this better...
+        self.KRATOSprint = self.procedures.KRATOSprint
```

Comment:
why not using the logger?

Reply:
This is old code. We can change all the print commands in a future PR

Reply:
It is actually using the Logger (KRATOSprint calls the Logger inside). However, for some project we needed to switch back to standard Python print, so I think that it's a good idea to have the print encapsulated in that function for the moment...

Reply:
Ok, I have added it after the 'if main' command as we agreed

Reply:
ah I see

as a side note, maybe this would be good to have for Kratos in general in the future
may I ask why you had to use the standard-python print?